accessible/toolboxview.component.js

```diff
@@ -0,0 +1,136 @@
+      </ol>
+    </li>
+    <div *ngIf="!toolboxHasCategories">
+      <toolbox-tree-view *ngFor="#block of getToolboxWorkspace(toolboxCategories[0]).topBlocks_; #i=index" [level]=1 [block]="block" [displayBlockMenu]="true" [clipboardService]="sharedClipboardService" [index]="i" [tree]="tree"></toolbox-tree-view>
+    </div>
+    </ol>
+  `,
+  directives: [blocklyApp.ToolboxTreeView],
+})
+.Class({
+  constructor: [blocklyApp.TreeService, function(_treeService) {
+    var _this = this;
+    var xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange = function() {
+      if (xhttp.readyState == 4 && xhttp.status == 200) {
+        _this.sightedToolbox = xhttp.responseXML;
+      }
+    };
+    xhttp.open('GET', blocklyApp.gameManager
+      .levelToolboxes[blocklyApp.gameManager.level], true);
+    xhttp.send();
+
+    this.toolboxCategories = [];
+    this.toolboxWorkspaces = {};
+    this.treeService = _treeService;
+
+    this.toolboxHasCategories = false;
+  }],
+  labelCategory: function(h2, i, tree) {
+    var parent = h2.parentNode;
+    while (parent != null && parent.tagName != 'LI') {
+      parent = parent.parentNode;
+    }
+    parent.setAttribute('aria-label', h2.innerText);
+    parent.id = 'toolbox-tree-node' + i;
+    if (i == 0 && tree.getAttribute('aria-activedescendant') ==
+      'toolbox-tree-node0') {
+      this.addClass(parent, 'blocklyActiveDescendant');
+      this.treeService.setActiveDesc(parent, tree.id);
+      parent.setAttribute('aria-selected', 'true');
+    }
+  },
+  addClass: function(node, classText) {
+    // ensure that node doesn't have class already in it
+    var classList = node.className;
+    if (classList) {
+      classList = classList.split(' ');
+    } else {
+      classList = [];
+    }
+    var canAdd = true;
+    for (var i = 0; i < classList.length; i++) {
+      if (classList[i] == classText) {
+        canAdd = false;
+      }
+    }
+    // add class if it doesn't
+    if (canAdd) {
+      if (classList.length) {
+        node.className += (' ' + classText);
+      } else {
+        node.className += classText;
+      }
+    }
+  },
+  makeArray: function(val) {
+    if (val) {
+      if (this.toolboxCategories.length) {
+        return this.toolboxCategories;
+      } else {
+        var categories = val.documentElement.getElementsByTagName('category');
+        if (categories.length) {
+          this.toolboxHasCategories = true;
+          this.toolboxCategories = Array.from(categories);
+          return this.toolboxCategories;
+        }
+        this.toolboxHasCategories = false;
+        this.toolboxCategories = Array.from(val.getElementsByTagName('xml'));
+        return this.toolboxCategories;
+      }
+    }
+  },
+  getToolboxWorkspace: function(categoryNode) {
+    if (categoryNode.attributes.name) {
+      var catName = categoryNode.attributes.name.value;
+    } else {
+      var catName = 'no-category';
+    }
+    if (this.toolboxWorkspaces[catName]) {
```

Comment:
FYI, I think "this.toolboxWorkspaces.hasOwnProperty(catName)" is the 'right' way to check object membership in JS, otherwise you'll miss cases where the value is, e.g., the empty string. Adding @NeilFraser to double check, since he's a JS master :)